sssd_test_framework/utils/tools.py

```diff
@@ -733,3 +740,81 @@ def __exec(self, cls, cmd: str, name: str | int, service: str | None = None) ->
             return None
 
         return cls.FromOutput(command.stdout)
+
+
+class SSHKeysUtils(MultihostUtility[MultihostHost]):
+    """
+    SSH key utilities on remote host.
+    """
+
+    def __init__(self, host: MultihostHost, fs: LinuxFileSystem) -> None:
+        super().__init__(host)
+        self.host: MultihostHost = host
+        self.fs: LinuxFileSystem = fs
+        self.user: str | None = None
+        self.path: str | None = None
+        self.file: str | None = None
+
+    def keygen(
+        self,
+        user: str,
+        path: str,
+        file: str = "id_rsa",
+        cipher: str = "rsa",
+        group: str | None = None,
+        args: list[Any] | None = None,
+    ) -> SSHKeysUtils:
+        """
+        Creates home directory and SSH key pair for given user.
+
+        :param user: Username.
+        :type user: str
+        :param path: Homedir directory.
+        :type path: str
+        :param file: SSH key file, defaults to "id_rsa"
+        :type file: str, optional
+        :param cipher: Encryption algorithm, defaults to "rsa"
+        :type cipher: str, optional
+        :param group: User group, defaults to None
+        :type group: str, optional
+        :param args: Additional arguments to pass to ssh-keygen, defaults to None
+        :type args: list[Any] | None
+        :return:
+        """
+
+        self.fs.backup("/home")
+        self.user = user
+        self.path = path
+        self.file = file
```

Comment:
This won't work if you call keygen multiple times.

You can:
* return (pubkey, privatekey) pair instead of self
* or return a new object e.g. `SSHKey` that will hold the information

Reply:
But SSHKeysUtils is not create multiple times, it is created only once for each test.

If you want it to be created multiple times, then you should do it via `@property`. It also should not inherit from MultihostUtility, but from object in this case.

```python
@property
def keygen(...) -> SSHKeyUtils:
    return SSHKeyUtils(...)

client.tools.keygen.generate()
```

Reply:
Removed the class inheritance and made keygen a property that calls SSHKeyUtils